lib/ecto/changeset.ex

```diff
@@ -1455,6 +1456,15 @@ defmodule Ecto.Changeset do
           end)
         end
 
+      query =
+        if is_list(opts) && opts[:prefix] do
```

Comment:
I think we can use `if prefix = opts[:prefix] do` and reuse the `prefix` variable later. Thoughts?

Reply:
Love that, thing is, the ```is_list``` check should go somewhere. Since you can use it like this:

```elixir
unsafe_validate_unique(changeset, [:city_name, :state_name], repo, "city must be unique within state")
```
Otherwise this test breaks: https://github.com/francocatena/ecto/blob/9f56ae90a1d19788d901d298c627d8314a04f5d4/test/ecto/changeset_test.exs#L1098

Personally I disagree with that, the message should always be set as an explicit ```message: "xx"``` option.

If we do ```if is_list(opts) && (prefix = opts[:prefix]) do``` (besides the ugliness) we get the warning 'the variable "prefix" is unsafe'.

Can we remove the "implicit" message conversion when a string is passed as an option? Is there another way to check it that I am not seeing?

Reply:
That test is wrong. :) It never triggers the error though because the validation doesn't fail. If the validation failed, it would error because elsewhere we expect opts to be a list. So you can assume that `opts` is a list and you can even add a guard to `unsafe_validate_unique`'s definition that guarantees so.

Reply:
Great, I've added the guard clause and the modified ```if``` statement. Now it looks better, tell me what you think =)